CONTRIBUTING.md

```diff
@@ -0,0 +1,63 @@
+# Contributing to Wallet-Provider
+Thank you for considering contributing to Wallet-Provider!
+
+## Reporting Bugs
+If you discover a bug, please create a [new issue](https://github.com/massalabs/wallet-provider/issues/new?assignees=&labels=issue%3Abug&template=bug.md&title=) on our GitHub repository.
+In your issue, please include a clear and concise description of the bug, any relevant code snippets, error messages, and steps to reproduce the issue.
+
+## Installation
+To start developing with wallet-provider, you must install all the necessary dev dependencies. You can do so by running the following command:
+
+```sh
+npm install
+```
+
+This will install all the required packages listed in the package.json file, allowing you to update, fix, or improve wallet-provider in any way you see fit. 
+
+## Contributing Code
+We welcome contributions in the form of bug fixes, enhancements, and new features.
+
+To contribute code, please follow these steps:
+
+1. Fork the wallet-provider repository to your own account.
+2. Create a new branch from the `main` branch for your changes.
+3. Make your changes and commit them to your branch.
+4. Push your branch to your fork.
+5. Create a pull request from your branch to the develop branch of the wallet-provider repository.
+
+> **NOTE:** When creating a pull request, please include a clear and concise title and description of your changes, as well as any relevant context or background information.
+
+
+## Tests
+Please ensure that your changes include any necessary tests.
+
+You can run the following command to run the tests:
+
+```sh
+npm run test
+```
+
+## Code Style
+Please ensure that your code follows the existing code style used in the project.
+We use the [MassaLabs Prettier configuration](https://github.com/massalabs/prettier-config-as) and [MassaLabs ESLint configuration](https://github.com/massalabs/eslint-config) for formatting and linting.
+
+You can run the following command to format your code before committing:
+
+```sh
+npm run fmt
+```
+
+
+## License
+By contributing to wallet-provider, you agree that your contributions will be licensed under the MIT License.
+
+## Documentation
+Wallet-provider provides complete documentation of all available functions and objects.
+
+To generate the documentation for a specific branch, run the following command:
+
+```sh
+npm run doc
+```
+
+The documentation will be generated in the `docs/documentation/html` directory.
```

Comment:
I think that this part should be in the readme file

Reply:
It is also in the README. Do you want me to delete it here ?

Reply:
Yes, because this documentation is targeting user of the library, not the dev (privateremarks are not published.